lib/util/stream_utils.js

```diff
@@ -515,17 +516,90 @@ shaka.util.StreamUtils = class {
             'for key', cacheKey);
         variant.decodingInfos.push(cache[cacheKey]);
       } else {
-        const result =
-            await navigator.mediaCapabilities.decodingInfo(decodingConfig);
-        cache[cacheKey] = result;
-        variant.decodingInfos.push(result);
+        // Do a final pass-over of the decoding config: if a given stream has
+        // multiple codecs, that suggests that it switches between those codecs
+        // at points of the go-through.
+        // mediaCapabilities by itself will report "not supported" when you
+        // put in multiple different codecs, so each has to be checked
+        // individually. So check each and take the worst result, to determine
+        // overall variant compatibility.
+        const results = await Promise.all(shaka.util.StreamUtils
+            .checkEachDecodingConfigCombination_(decodingConfig));
+        /** @type {?MediaCapabilitiesDecodingInfo} */
+        let overallResult = null;
+        for (const result of results) {
+          if (!overallResult) {
+            overallResult = result;
+          } else {
+            // Sadly, out compiler is out of date and does not recognize the
```

Comment:
Why not update the compiler?

Reply:
It's been a while, but if I remember correctly we stopped updating the compiler because there was some bug with it that they refused to fix?